### PR TITLE
Add gnomAD mitochondrial frequencies

### PR DESCRIPTION
**Pull request by bjhall**
Solves https://github.com/Clinical-Genomics/scout/issues/2454

This PR adds support for mitochondrial gnomAD frequencies. It can parse them from VEP's CSQ string or normal INFO fields. Either way, the fields should be called: `GNOMAD_MT_AF_HOM` and `GNOMAD_MT_AF_HET`

The frequency parsing/building/viewing in Scout is very messy (due to legacy stuff), and this adds even more complexity. However I've been careful not to disrupt anything, and have done fairly extensive testing to make sure everything is shown correctly.

It should change nothing for non-mitochondrial variants. And for mitochondrial variants it will show this instead:

Variant list:
bild

Variant details:
bild

### CODE REVIEW

**Comment by northwestwitch** [comment]
I can review @bjhall!

File: scout/parse/variant/frequency.py [hunk 1]
```diff
@@ -22,6 +22,7 @@ def parse_frequencies(variant, transcripts):
     # Gnomad have both snv and sv frequencies
     gnomad_keys = ["GNOMADAF", "GNOMAD_AF", "gnomad_svAF"]
     gnomad_max_keys = ["GNOMADAF_popmax", "GNOMADAF_POPMAX", "GNOMADAF_MAX"]
+    gnomad_mt_keys = ["GNOMAD_MT_AF_HOM", "GNOMAD_MT_AF_HET"]
```
Comment (northwestwitch) [inline scout/parse/variant/frequency.py:25]: 👍

File: scout/parse/variant/frequency.py [hunk 2]
```diff
@@ -59,6 +60,15 @@ def parse_frequencies(variant, transcripts):
             frequencies["gnomad_max"] = gnomad_max
             break
 
+    # For mitochondrial variants, keep both "hom" and "het" freqs
+    gnomad_mt_hom = parse_frequency(variant, "GNOMAD_MT_AF_HOM")
+    if gnomad_mt_hom is not None:
```
Comment (northwestwitch) [inline scout/parse/variant/frequency.py:65]:
```suggestion
    if gnomad_mt_hom:
```

File: scout/parse/variant/frequency.py [hunk 3]
```diff
@@ -59,6 +60,15 @@ def parse_frequencies(variant, transcripts):
...
+        frequencies["gnomad_mt_homoplasmic"] = gnomad_mt_hom
+
+    gnomad_mt_het = parse_frequency(variant, "GNOMAD_MT_AF_HET")
+    if gnomad_mt_het is not None:
```
Comment (northwestwitch) [inline scout/parse/variant/frequency.py:69]:
```suggestion
    if gnomad_mt_het:
```

File: scout/parse/variant/transcript.py [hunk 4]
```diff
@@ -154,6 +154,12 @@ def parse_transcripts(raw_transcripts, allele=None):
                 float(fractmatch) for fractmatch in superdups_fractmatch.split("&")
             ]
 
+        # Get mitochondrial gnomAD frequencies
+        if "GNOMAD_MT_AF_HOM" in entry and entry.get("GNOMAD_MT_AF_HOM") != "":
```
Comment (northwestwitch) [inline scout/parse/variant/transcript.py:158]:
```suggestion
        if entry.get("GNOMAD_MT_AF_HOM", "") != "":
```

File: scout/parse/variant/transcript.py [hunk 5]
```diff
@@ -154,6 +154,12 @@ def parse_transcripts(raw_transcripts, allele=None):
...
+            transcript["gnomad_mt_homoplasmic"] = float(entry.get("GNOMAD_MT_AF_HOM"))
+        if "GNOMAD_MT_AF_HET" in entry and entry.get("GNOMAD_MT_AF_HET") != "":
```
Comment (northwestwitch) [inline scout/parse/variant/transcript.py:160]:
```suggestion
        if entry.get("GNOMAD_MT_AF_HET", "") != "":
```

**Review by northwestwitch — APPROVED** [review]
Nice new feature, and looks like it's working perfectly from your tests!

I've got a few (untested) small suggestions that you can accept if you want, but otherwise ot's fine to merge I think!

File: scout/parse/variant/frequency.py (hunk 2 above)
Comment (bjhall) [reply scout/parse/variant/frequency.py:65]: I was just sticking to the theme. Every other place in this file uses `is not None`. I could change it for all of them though, but I don't feel like rocking the boat too much :)

File: scout/parse/variant/frequency.py (hunk 3 above)
Comment (bjhall) [reply scout/parse/variant/frequency.py:69]: Same as above

File: scout/parse/variant/frequency.py (hunk 2 above)
Comment (northwestwitch) [reply scout/parse/variant/frequency.py:65]: Ok, but then let's stick to is not None, as it's also less prone to give unwanted results!

**Pull request merged**